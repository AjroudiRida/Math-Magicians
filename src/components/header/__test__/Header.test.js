import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Header from '../Header';

const HeaderElement = () => (
  <BrowserRouter>
    <Header />
  </BrowserRouter>
);

it('should renders the header components in the document', () => {
  render(<HeaderElement />);
  const navigationElement = screen.getByRole('navigation');
  expect(navigationElement).toBeInTheDocument();
});

it('should be visible in the document', () => {
  render(<HeaderElement />);
  const navigationElement = screen.getByRole('navigation');
  expect(navigationElement).toBeVisible();
});

it('should match the Dom snapshot', () => {
  const headerTree = renderer.create(<HeaderElement />).toJSON();
  expect(headerTree).toMatchSnapshot();
});
