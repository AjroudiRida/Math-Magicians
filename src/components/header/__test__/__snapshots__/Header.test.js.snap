// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the Dom snapshot 1`] = `
<header
  className="header"
>
  <nav
    className="nav"
  >
    <a
      className="logo"
      href="/"
      onClick={[Function]}
    >
      Math Magicians
    </a>
    <ul
      className="menu"
    >
      <li
        className="menu_item"
      >
        <a
          className="link_item"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="menu_item"
      >
        <a
          className="link_item"
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        className="menu_item"
      >
        <a
          className="link_item"
          href="/quote"
          onClick={[Function]}
        >
          Quotes
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
